src/core_read.cpp

```diff
@@ -193,14 +193,13 @@ bool DecodePSBT(PartiallySignedTransaction& psbt, const std::string& base64_tx,
     return true;
 }
 
-uint256 ParseHashStr(const std::string& strHex, const std::string& strName)
+bool ParseHashStr(const std::string& strHex, uint256& result)
 {
-    if (!IsHex(strHex)) // Note: IsHex("") is false
-        throw std::runtime_error(strName + " must be hexadecimal string (not '" + strHex + "')");
+    if (!IsHex(strHex) || (strHex.size() != 64))
```

Comment:
Nit: Redundant parentheses around `strHex.size() != 64` :-)

Reply:
Could go either way but I like explicit precedence to guide the eye.

Reply:
Perhaps swap the two boolean conditions? If the condition `strHex.size()` is false then the function can return without having to evaluate the condition `IsHex(strHex)`

```c++
if ((strHex.size() != 64) || !IsHex(strHex)) return false;
```